core/whistle_services-1.0.0/src/whs_account_conversion.erl

```diff
@@ -0,0 +1,174 @@
+            lager:debug("unable to determin descendants of ~s: ~p", [AccountId, _R]),
+            Error;
+        {'ok', JObjs} ->
+            _ = [set_reseller_id(ResellerId, SubAccountId)
+                 || JObj <- JObjs
+                        ,(SubAccountId = wh_doc:id(JObj)) =/= AccountId
+                ],
+            'ok'
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Set teh reseller_id to the provided value on the provided account
+%% @end
+%%--------------------------------------------------------------------
+-spec set_reseller_id(ne_binary(), ne_binary()) -> {'error', _} | 'ok'.
+set_reseller_id(Reseller, Account) ->
+    AccountId = wh_util:format_account_id(Account, 'raw'),
+    ResellerId = wh_util:format_account_id(Reseller, 'raw'),
+    io:format("setting account ~s reseller id to ~s~n", [AccountId, ResellerId]),
+    _ = update_account_definition(AccountId, <<"pvt_reseller_id">>, ResellerId),
+    maybe_update_services(AccountId, <<"pvt_reseller_id">>, ResellerId).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_update_services(ne_binary(), ne_binary(), any()) -> {'error', _} | 'ok'.
+maybe_update_services(AccountId, Key, Value) ->
+    case couch_mgr:open_doc(?WH_SERVICES_DB, AccountId) of
+        {'error', _R}=Error ->
+            io:format("unable to open services doc ~s: ~p~n", [AccountId, _R]),
+            Error;
+        {'ok', JObj} ->
+            case couch_mgr:save_doc(?WH_SERVICES_DB, wh_json:set_value(Key, Value, JObj)) of
+                {'ok', _} -> 'ok';
+                {'error', _R}=Error ->
+                    io:format("unable to set ~s on services doc ~s: ~p~n", [Key, AccountId, _R]),
+                    Error
+            end
+    end.
+
+-spec update_account_definition(ne_binary(), ne_binary(), any()) -> {'error', _} | 'ok'.
+update_account_definition(AccountId, Key, Value) ->
+    AccountDb = wh_util:format_account_id(AccountId, 'encoded'),
+    case couch_mgr:open_doc(AccountDb, AccountId) of
+        {'error', _R}=Error ->
+            io:format("unable to open account ~s defintion: ~p~n", [AccountId, _R]),
+            Error;
+        {'ok', JObj} ->
+            case couch_mgr:save_doc(AccountDb, wh_json:set_value(Key, Value, JObj)) of
+                {'ok', NewJObj} ->
+                    _ = couch_mgr:ensure_saved(?WH_ACCOUNTS_DB, NewJObj),
+                    'ok';
+                {'error', _R}=Error ->
+                    io:format("unable to set pvt_reseller on account ~s defintion: ~p~n", [AccountId, _R]),
+                    Error
+            end
+    end.
+
+-spec has_reseller_descendants(ne_binary()) -> boolean().
+has_reseller_descendants(AccountId) ->
+    %% its very important that this check not operate against stale data!
+    _ = couch_mgr:flush_cache_docs(),
```

Comment:
This is very heavy-handed...surely `couch_mgr:flush_cache_docs(?WH_ACCOUNTS_DB)` would be specific enough?

Also, the cache isn't involved in view lookups, so not sure why you're flushing here anyway?